dashdblocal_notebooks/livy-notebooks/IoT_Demo_Pyspark.ipynb

```diff
@@ -0,0 +1,1246 @@
+    "    df = sqlsc.read.json(rdd2, schema)\n",
+    "    df.select([\"payload.temperature\", \"payload.tempOutside\", \"payload.powerProd\", \"payload.noiseLevel1\", \"payload.time\", \"deviceId\", \"deviceType\", \"eventType\"]).show()\n",
+    "\n",
+    "runStreamingContext(processStream1)   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Write IoT records to a persistent table\n",
+    "\n",
+    "In the desired deployed application we want to write the records read from the queue into a Db2 table instead of only printing out the datasets. In this demonstration here we will always first *empty* the table in the first batch after the application is started. We use Spark's *'Overwrite'* mode for writing the first batch to clear a pre-existing table and then use *'Append'* mode in subsequent batches.\n",
+    "\n",
+    "You can check in your Db2 Warehouse console that a table called USERDATA.IOT_EVENTS has been created. For example, run the following SQL query to see how many records have been written: \n",
+    "> SELECT COUNT(*) FROM USERDATA.IOT_EVENTS.\n",
+    "\n",
+    "You will notice that there are only a few rows compared to the number of generated records. For example, with a timeout set to 10s, you will collect approximately 8 rows. With a timeout of 50s, there will be approximetely 30 etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database"
+     ]
+    }
+   ],
+   "source": [
+    "tableName= \"USERDATA.IOT_EVENTS\"\n",
+    "savemode = 'overwrite'\n",
+    "\n",
+    "def processStream2(rdd):\n",
+    "    global savemode\n",
+    "    rdd2 = rdd.map(lambda x: x[1])\n",
+    "    df = sqlsc.read.json(rdd2, schema)\n",
+    "    df2=df.select([\"payload.temperature\", \\\n",
```

Comment:
Nitpicking: Python doesn't require trailing backslashes (`\`) in list definitions,

```python
    df2=df.select(["payload.temperature",
               "payload.tempOutside",
               "payload.powerProd",
               "payload.noiseLevel1",
               "payload.time",
               "deviceId",
               "deviceType",
               "eventType"])
```

should work just fine.

Reply:
ok :)